propagators/aws/xray/Propagator.php

```diff
@@ -0,0 +1,151 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace Propagators\Aws\Xray;
+
+use OpenTelemetry\Trace as API;
+use OpenTelemetry\Sdk\Trace\SpanContext;
+
+/**
+ * Implementation of AWS Trace Header Protocol
+ * See below: 
+ * https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader
+ * 
+ * Propagator serializes Span Context to/from AWS X-Ray headers.
+ * Example AWS X-Ray format:
+ * 
+ * X-Amzn-Trace-Id: Root={traceId};Parent={parentId};Sampled={samplingFlag}
+ * X-Amzn-Trace-Id: Root=1-5759e988-bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=1
+ */
+class AwsXrayPropagator implements API\TextMapFormatPropagator
+{
+    public const AWSXRAY_TRACE_ID_HEADER = 'X-Amzn-Trace-Id';
+    private const VERSION_NUMBER = '1';
+    private const TRACE_HEADER_DELIMITER = ';';
+    private const KV_DELIMITER = '=';
+
+    private const TRACE_ID_KEY = 'Root';
+    private const TRACE_ID_LENGTH = 35;
+    private const TRACE_ID_VERSION = '1';
+    private const TRACE_ID_DELIMITER = '-';
+    private const TRACE_ID_DELIMITER_INDEX_1 = 1;
+    private const TRACE_ID_DELIMITER_INDEX_2 = 10;
+    private const TRACE_ID_TIMESTAMP_LENGTH = 8;
+    private const TRACE_ID_RANDOMHEX_LENGTH = 24;
+
+    private const PARENT_ID_KEY = 'Parent';
+    private const SPAN_ID_LENGTH = 16;
+    
+    private const SAMPLED_FLAG_KEY = 'Sampled';
+    private const IS_SAMPLED = '1';
+    private const NOT_SAMPLED = '0';
+    
+    /**
+     * Returns list of fields used by HTTPTextFormat
+     */
+    public static function fields(): array
+    {
+        return [self::AWSXRAY_TRACE_ID_HEADER];
+    }
+
+    /**
+     * Creates and injects the traceHeader
+     * X-Amzn-Trace-Id: Root={traceId};Parent={parentId};Sampled={samplingFlag}
+     * X-Amzn-Trace-Id: Root=1-5759e988-bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=1
+     */
+    public static function inject(API\SpanContext $context, &$carrier, API\PropagationSetter $setter): void
+    {
+        $traceId = $context->getTraceId();
+        $spanId = $context->getSpanId();
+
+        if (!self::isValidSpanId($spanId) || !self::isValidTraceId($traceId)) return;
+
+        $timestamp = substr($traceId, 0, self::TRACE_ID_TIMESTAMP_LENGTH);
+        $randomHex = substr($traceId, self::TRACE_ID_TIMESTAMP_LENGTH);
+        $samplingFlag = $context->isSampled() ? self::IS_SAMPLED : self::NOT_SAMPLED;
+
+        $traceHeader = 'Root=' . self::VERSION_NUMBER . '-' . $timestamp . "-" . $randomHex . ";Parent=" . $spanId. ";Sampled=" . $samplingFlag;
```

Comment:
`-` and `;` are already defined above, should better use them. Also, why use two types of quotation marks `' '` and `" "` here?